app/views/shared/_stats.html.erb

```diff
@@ -0,0 +1,17 @@
+<% @user ||= current_user %>
+
+<div class="stats">
+  <a href="<%= following_user_path(@user) %>">
+    <strong id="following" class="stat">
+      <%= @user.following.count %>
```

Comment:
phân biệt count, size, length

Reply:
count: count dùng câu lệnh sql (SELECT COUNT() FROM…) truy vấn số lượng activerecord. count không được lưu trữ trong đối tượng được gọi, nghĩa là một lần gọi count thì nó sẽ thực hiện một truy vấn (SELECT COUNT() FROM…) vào database dù cho trước đó đã gọi hay chưa.
    size: khi object đã đuợc load, size sẽ thực hiện trên chính object đó trong bộ nhớ (không truy vấn vào database, tuơng tự length ở lần thứ 2 trở đi). Còn khi object đó chưa được load, nó sẽ thực hiện câu lệnh (SELECT COUNT(*) FROM…). Sử dụng size là tốt nhất
    length: length đếm số lượng mà không sử dụng thêm câu truy vấn nếu đối tượng đó đã được nạp, ngược lại, nó sẽ nạp toàn bộ object vào bộ nhớ và thực hiện đếm. Nhưng khi gọi length lần thứ 2 trở đi, lenght sẽ không còn load từ database nữa, mà trả về luôn kết quả (đã được lưu trong bộ nhớ).